src/map.cpp

```diff
@@ -5083,15 +5083,11 @@ std::list<item> map::use_charges(const tripoint &origin, const int range,
         // if vehicle has FAUCET can use contents from any of the tanks
         if( v->has_part( "FAUCET" ) ) {
             for( auto &pt : v->parts ) {
-                if( pt.is_tank() ) {
-                    for( const auto &obj : pt.contents() ) {
-                        if( obj.typeId() == type ) {
-                            ret.push_back( pt.drain( quantity ) );
-                            quantity -= ret.back().charges;
-                            if( quantity == 0 ) {
-                                return ret;
-                            }
-                        }
+                if( pt.is_tank() && pt.ammo_current() == type ) {
```

Comment:
True. `ammo_current()` does, too.

The loop iterates over all vehicle parts. IMO the check is good for readability.

Also, if `ammo_current()` returned a matching type for something that is not a tank (e.g. a turret, engine, funnel, hydroponic unit, - anything in the future that could have a matching `ammo` due to implementation details), this could result in breakage. The intent is to craft from tanks only.

Should I add a comment to this effect?